src/fstab-generator/fstab-generator.c

```diff
@@ -358,6 +358,15 @@ static int add_mount(
                 "Documentation=man:fstab(5) man:systemd-fstab-generator(8)\n",
                 source);
 
+        if (strneq(fstype, "nfs", 3) && !automount &&
```

Comment:
what's the rationale for this fstype check? It appears overly generic to match against all fstypes that start with "nfs", no? There might be "nfs-ng" one day, who knows, and it might take completely different settings...

Hence, I think it would be a good idea to use `STR_IN_SET(fstype, "nfs", "nfs4")` here, as that appears to be what you actually want to check for, no?